src/test/scala/LiveVarsAnalysisTests.scala

```diff
@@ -0,0 +1,350 @@
+          LocalAssign(R1, constant1),
+          call("callee", Some("wrapper1_second_call"))
+        ),
+        block("wrapper1_second_call",
+          call("callee2", Some("wrapper1_return"))),
+        block("wrapper1_return", ret)
+      ),
+      proc("wrapper2",
+        block("wrapper2_first_call",
+          LocalAssign(R2, constant1),
+          call("callee", Some("wrapper2_second_call"))
+        ),
+        block("wrapper2_second_call",
+          call("callee3", Some("wrapper2_return"))),
+        block("wrapper2_return", ret)
+      )
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+
+    val liveVarAnalysisResults = InterLiveVarsAnalysis(program).analyze()
+    val blocks = program.blocks
+    assert(liveVarAnalysisResults(blocks("wrapper1_first_call").jump) == Map(R1 -> Top, r30 -> Top))
+    assert(liveVarAnalysisResults(blocks("wrapper2_first_call").jump) == Map(R2 -> Top, r30 -> Top))
+
+  }
+
+  def deadBeforeCall(): Unit = {
+    import IRDSL._
+
+    val r30 = Register("R30", BitVecType(64))
+    var program = prog(
+      proc("main",
+        block("lmain",
+          call("killer", Some("aftercall"))
+        ),
+        block("aftercall",
+          LocalAssign(R0, R1),
+          ret
+        )
+      ),
+      createSimpleProc("killer", Seq(LocalAssign(R1, bv64(1))))
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+
+    val liveVarAnalysisResults = InterLiveVarsAnalysis(program).analyze()
+    val blocks = program.blocks
+
+    assert(liveVarAnalysisResults(blocks("aftercall")) == Map(R1 -> Top, r30 -> Top))
+    assert(liveVarAnalysisResults(blocks("lmain")) == Map(r30 -> Top))
+  }
+
+  def simpleBranch(): Unit = {
+    import IRDSL._
+
+    val r30 = Register("R30", BitVecType(64))
+    val r1Assign = new LocalAssign(R0, R1, Some("00001"))
+    val r2Assign = new LocalAssign(R0, R2, Some("00002"))
+
+    var program : Program = prog(
+      proc(
+        "main",
+        block(
+          "lmain",
+          goto("branch1", "branch2")
+        ),
+        block(
+          "branch1",
+          r1Assign,
+          goto("main_return")
+        ),
+        block(
+          "branch2",
+          r2Assign,
+          goto("main_return")
+        ),
+        block("main_return", ret)
+      )
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+
+    val blocks = program.blocks
+    val liveVarAnalysisResults = InterLiveVarsAnalysis(program).analyze()
+
+    assert(liveVarAnalysisResults(blocks("branch1")) == Map(R1 -> Top, r30 -> Top))
+    assert(liveVarAnalysisResults(blocks("branch2")) == Map(R2 -> Top, r30 -> Top))
+    assert(liveVarAnalysisResults(blocks("lmain")) == Map(R1 -> Top, R2 -> Top, r30 -> Top))
+
+  }
+
+  def recursionInfinite(): Unit = { // can't handle this infinite recursion case
```

Comment:
This just fails because the live variable doesnt propagate back over the infinite recursion right? The test currently doesn't have an assertion so passes trivially, it would be better to have the appropriate assertions and disable the test with the `ignore` flag. Its also ok to leave it here failing so we know its an issue.